shell/browser/notifications/win/windows_toast_notification.cc

```diff
@@ -636,4 +1249,282 @@ IFACEMETHODIMP ToastEventHandler::Invoke(
   return S_OK;
 }
 
+struct CoTaskMemStringTraits {
+  typedef PWSTR Type;
+
+  inline static bool Close(_In_ Type h) throw() {
+    ::CoTaskMemFree(h);
+    return true;
+  }
+
+  inline static Type GetInvalidValue() throw() { return nullptr; }
+};
+typedef HandleT<CoTaskMemStringTraits> CoTaskMemString;
+
+// #pragma GCC diagnostic push
+// #pragma GCC diagnostic ignored "-Wmissing-braces"
+// #pragma GCC diagnostic ignored "-Wc++98-compat-extra-semi"
+//  For the app to be activated from Action Center, it needs to provide a COM
+//  server to be called when the notification is activated.  The CLSID of the
+//  object needs t  registered with the OS via its shortcut so that it knows
+//  who to call later.
+class NotificationActivator final
+    : public RuntimeClass<RuntimeClassFlags<ClassicCom>,
+                          INotificationActivationCallback> {
+ public:
+  HRESULT STDMETHODCALLTYPE
+  Activate(_In_ LPCWSTR appUserModelId,
+           _In_ LPCWSTR invokedArgs,
+           _In_reads_(dataCount) const NOTIFICATION_USER_INPUT_DATA* data,
+           ULONG dataCount) override {
+    // packing user input structs into stream
+    std::wstringstream stm;
+    // feat: Refine notification-activation for cold start
+    if (dataCount) {
+      stm << L"[";  // json array open brace
+      std::for_each(data, data + dataCount,
+                    [&](const NOTIFICATION_USER_INPUT_DATA& item) {
+                      stm << item
+                          // avoid problem with last delimeter
+                          << std::wstring((static_cast<ULONG>(&item - data +
+                                                              1) == dataCount)
+                                              ? L""
+                                              : L",");
+                    });
+      stm << L"]";  // json array close brace
```

Comment:
This is wildly unsafe, please use an appropriate object builder or JSON implementation from `//base` or `//v8`

Reply:
Hello @MarshallOfSound  this is a vary very short list of sel formed JSON into current project
base\trace_event\log_message.cc, Ln 23
jingle\notifier\listener\notification_defines.cc, Ln 75
base\test\trace_to_file.cc, Ln 64
**Especially logger which is used widely.** I know what I'm doing and I want to exclude unnecessary dependencies such as JSON related classes.
I have simple question: Why does we have self formed JSON into logger in case if my 3 lines are so significant? 
I'm asking my question as developer and not with purpose to arguing. Think You agree with me that "better simpler than clever", not?

Reply:
> Have simple question: Why does we have self formed JSON into logger in case if my 3 lines of self formed JSON are so significant?

All the code you referenced is deep in Chromium not in electron code.  This code / logic in it's current state is left up to users to consume and it is our responsibility as maintainers to ensure that users are safe and secure, that takes priority over any theoretical performance differences or "visions".

Manually forming JSON with user data is **unsafe**.  We have no way to guarantee their aren't quotes, escape sequences, etc. in that data.

To be clear, in it's current state this code will not be landed.

Reply:
`We have no way to guarantee their aren't quotes, escape sequences, etc. in that data.` - for this concrete part we have such guaranties
Json forming was modified in fca8ff5a8c978fb7c3e2a7acf8216ee359e4d7b1, my remarks above